src/Handler/GuzzleV6/GuzzleHandler.php

```diff
@@ -40,20 +41,40 @@ public function __invoke(Psr7Request $request, array $options = [])
                 . ' ' . \GuzzleHttp\default_user_agent()
         );
 
-        return $this->client->sendAsync($request, $options)->otherwise(
-            static function (\Exception $e) {
-                $error = [
-                    'exception'        => $e,
-                    'connection_error' => $e instanceof ConnectException,
-                    'response'         => null,
-                ];
+        return $this->client->sendAsync($request, $this->parseOptions($options))
+            ->otherwise(
+                static function (\Exception $e) {
+                    $error = [
+                        'exception'        => $e,
+                        'connection_error' => $e instanceof ConnectException,
+                        'response'         => null,
+                    ];
 
-                if ($e instanceof RequestException && $e->getResponse()) {
-                    $error['response'] = $e->getResponse();
+                    if ($e instanceof RequestException && $e->getResponse()) {
+                        $error['response'] = $e->getResponse();
+                    }
+
+                    return new Promise\RejectedPromise($error);
                 }
+            );
+    }
 
-                return new Promise\RejectedPromise($error);
-            }
-        );
+    private function parseOptions(array $options)
+    {
+        if (isset($options['__on_transfer_stats'])) {
```

Comment:
Currently, no layer of the SDK is setting `on_stats`, though users of the Guzzle 6 handler are free to do so. `__on_transfer_stats` is set up as an option that is understood by both Guzzle handlers, and it has a different name so that no user code that is currently collecting Guzzle 6 stats would be affected.